platform/android/java/lib/src/org/godotengine/godot/utils/CommandLineFileParser.kt

```diff
@@ -0,0 +1,88 @@
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+package org.godotengine.godot.utils
+
+import java.io.InputStream
+import java.nio.charset.StandardCharsets
+import java.util.ArrayList
+
+/**
+ * A class that parses the content of file storing command line params. Usually, this file is saved
+ * in `assets/_cl_` on exporting an apk
+ *
+ * Returns a list of command lines
+ */
+class CommandLineFileParser {
+	fun parseCommandLine(inputStream: InputStream): List<String> {
+		return try {
+			val headerBytes = ByteArray(4)
+			var argBytes = inputStream.read(headerBytes)
+			if (argBytes < 4) {
+				return listOf()
+			}
+			val argc =
+				(headerBytes[3].toInt() and 0xFF) shl 24 or
+				((headerBytes[2].toInt() and 0xFF) shl 16) or
+				((headerBytes[1].toInt() and 0xFF) shl 8) or
+				(headerBytes[0].toInt() and 0xFF)
+
+			val cmdline = ArrayList<String>(argc)
+			for (i in 0 until argc) {
+				// reset before reading another header
+				headerBytes.fill(0)
```

Comment:
This is new, and I'm not sure about it, but I have a feeling that we should reset the buffer before reading the next header, in case the new header leads to a shorter string, and so we don't want the old one still partially occupying the array... 🤔

Reply:
That should be taken care of by checking that `argBytes < 4`; `Inputstream#read(...)` returns the number of bytes it read, and in this scenario we expect 4 bytes describing how long the next String will be and thus fully filling up the `headerBytes` byte array.

Are you able to validate the scenario you described using the unit tests?

Reply:
> Are you able to validate the scenario you described using the unit tests?

No, can't think of an edge case like that atm, which wouldn't be cought by `argBytes < 4`  check

Removed the extra line, thanks